crates/ruff_linter/src/rules/pydoclint/rules/check_docstring.rs

```diff
@@ -711,6 +732,28 @@ fn yields_documented(
         || (matches!(convention, Some(Convention::Google)) && starts_with_yields(docstring))
 }
 
+/// Returns the first subscript element of a generator annotation, if it exists.
+fn generator_yield_annotation<'a>(expr: &'a Expr, checker: &'a Checker) -> Option<&'a Expr> {
+    let slice = expr.as_subscript_expr()?.slice.as_tuple_expr()?;
```

Comment:
Unfortunately while `Generator[int, str, None]` has an `Expr::Tuple(_)` inside the `Generator[]` slice, `Iterator[Any]` has an `Expr::Name(_)` inside the `Iterator` slice. So the logic you're adding currently works for functions returning `Generator[]` but not those returning `Iterator[]`. I'm working on a fix for this and a few more edge cases I spotted; I'll push them to your PR branch

Reply:
Well spotted, thank you!